src/extension.ts

```diff
@@ -49,14 +49,16 @@ export function activate(ctx: sourcegraph.ExtensionContext = DUMMY_CTX): void {
     )
 
     ctx.subscriptions.add(
-        reregisterWhenEnablementChanges('basicCodeIntel.hover', () =>
+        reregisterWhenEnablementChanges('basicCodeIntel.hover', true, () =>
             sourcegraph.languages.registerHoverProvider(DOCUMENT_SELECTOR, {
                 provideHover: (doc, pos) => {
-                    if (
-                        !sourcegraph.configuration.get<Settings>().value[
-                            'basicCodeIntel.hover'
-                        ]
-                    ) {
+                    let value = sourcegraph.configuration.get<Settings>().value[
+                        'basicCodeIntel.hover'
+                    ]
+                    if (value === undefined) {
+                        value = true
+                    }
+                    if (!value) {
```

Comment:
You could shorten this to `if (sourcegraph.configuration.get<Settings>().value['basicCodeIntel.hover'] !== false)`